### PR TITLE
Use WC helper function to get card brand label from Card type on My Account Subscription page

### LINKED ISSUE #3518: Subscription Payment method uses lowercase 'visa' and 'mastercard'
**Describe the bug**
On the **My Account > Subscription page**, the subscription payment method's card brand is displayed directly from the Stripe payment method object and that results in it being displayed in all lowercase:

| Mastercard | Visa | JCB |
|--------|--------|--------|
| Image|Image |Image| 

**To Reproduce**
1. Purchase a new subscription using a Visa, Mastercard, JCB or any card brand.
   - You can find the full list here: https://docs.stripe.com/testing#cards
2. View the subscription on the **My Account → Subscription** page.
3. You should notice that the brand is always displayed in lowercase. 

**Expected behavior**
The brand should use the correct capitalization.

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #3518 

## Changes proposed in this Pull Request:

On the **My Account → Subscription** page, we display details about the subscription payment method. In this label, we would display the card brand directly returned from Stripe and that would be all lowercase. 

This PR fixes that by making sure we use the correct capitalisation. 

| Before | After |
|--------|--------|
| Screenshot 2024-10-17 at 3 53 13 pm | Screenshot 2024-10-17 at 3 52 36 pm | 

> [!note]
> WC Core's `wc_get_credit_card_type_label()` helper function still uses the old capitalization styling of Mastercard (WC core displays MasterCard -- uppercase C). I've submitted https://github.com/woocommerce/woocommerce/pull/52101 to fix that. 

## Testing instructions

1. Purchase a new subscription using a Visa, Mastercard, JCB or any card brand.
   - You can find the full list here: https://docs.stripe.com/testing#cards
2. View the subscription on the My Account → Subscription page.
3. On `develop` all card brands are displayed in lower case, on this brand, the correct brand capitalisation should be applied. 

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

**Review by annemirasol — APPROVED** [review]
Tests out well, LGTM 🚢 

> I've submitted https://github.com/woocommerce/woocommerce/pull/52101 to fix that.

Nice 👏  I noticed during testing that Diners Club displays as `Diners`, and UnionPay displays as `Unionpay`. Just FYI -- in case you think it's worth including changes for these in your other PR.

File: tests/phpunit/test-wc-stripe-payment-gateway.php [hunk 1]
```diff
@@ -568,4 +568,61 @@ public function test_get_balance_transaction_id_from_charge() {
...
+
+		set_transient( $transient_key, [ $mock_payment_method ], DAY_IN_SECONDS );
+		$this->assertEquals( 'Via Visa card ending in 4242', $this->gateway->maybe_render_subscription_payment_method( 'N/A', $mock_subscription ) );
+
+		// MasterCard ending in 1234
+		$mock_payment_method->card->brand = 'mastercard';
+		$mock_payment_method->card->last4 = '1234';
+
+		set_transient( $transient_key, [ $mock_payment_method ], DAY_IN_SECONDS );
+		$this->assertEquals( 'Via MasterCard card ending in 1234', $this->gateway->maybe_render_subscription_payment_method( 'N/A', $mock_subscription ) );
```
Comment (annemirasol) [inline tests/phpunit/test-wc-stripe-payment-gateway.php:605]: Noting that https://github.com/woocommerce/woocommerce/pull/52101 will break this. I suppose we just wait until the changes land and update this when it breaks?

**Review by diegocurbelo — APPROVED** [review]
Code changes look good, and it tests well!!

As @annemirasol mentioned, we'll need to update the unit test after [#52101](https://github.com/woocommerce/woocommerce/pull/52101) is merged, but we can handle that in a "fix compatibility tests" PR like we did in the past.

**Comment by james-allan** [comment]
> I noticed during testing that Diners Club displays as Diners, and UnionPay displays as Unionpay. Just FYI -- in case you think it's worth including changes for these in your other PR.

Thanks @annemirasol. I wasn't able to find much information on Diners Club and why WC would just display it as "Diners". As far as I can tell, that's the colloquial/shorthand term but its never had that officially.

UnionPay was never Unionpay either as far as I can tell so I'll get those fixed up in that PR too.

edit: The problem with UnionPay is that WC doesn't have any UnionPay case at all.

**Pull request merged**